### PR TITLE
Add refile support

### PR DESCRIPTION
**Pull request by omoshetech-t**
Add support for a modern file upload library. https://github.com/refile/refile

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -47,6 +47,9 @@ group :test do
   gem 'paperclip', ['>= 3.4', '!= 4.3.0']
   gem 'poltergeist', '~> 1.5'
   gem 'rack-cache', require: 'rack/cache'
+  gem 'refile', '>= 0.5.5', require: 'refile/rails'
+  gem 'refile-mini_magick', '>= 0.1.0'
+  gem 'refile-mongoid', '>= 0.0.1'
```
Comment (mshibuya) [inline Gemfile]: `refile-mongoid` should go into `:mongoid` group, since it is only required for mongoid builds.

**Comment by mshibuya** [comment]
CI build is failing, can you investigate?

**Comment by omoshetech-t** [comment]
> `refile-mongoid` should go into `:mongoid` group, since it is only required for mongoid builds.
I'll fix it.

> CI build is failing, can you investigate?
I forgot to check required Ruby version of the refile gem. It needs >= 2.1.0. This is one reason of build failure. So is it difficult to merge this request?

Build failures also occurred when satisfy refile required Ruby version. I can investigate reasons of this failures.

**Comment by mshibuya** [comment]
> I forgot to check required Ruby version of the refile gem. It needs >= 2.1.0. This is one reason of build failure.

Then you can conditionally install(and test) refile gem for Ruby >= 2.1.0. To merge in, passing all builds is mandatory.

**Comment by omoshetech-t** [comment]
I add conditional statements of Ruby version for build failures fix.

The build failures when satisfy refile required Ruby version, cause sass-rails version. 
sass-rails 5.0.0 is installed when I add `gem "refile", ">= 0.5.5"` to rails_4.2.gemfile, but this project needs sass-rails >= 5.0.2. If I don't add `gem "refile", ">= 0.5.5"`, sass-rails 5.0.3 is installed. This problem doesn't occur bundler version >= 1.9.0.pre. rails_4.0.gemfile and rails_4.1.gemfile also have sass-rails version problem but they are a little different. In this case, sass-rails 4.0.0 is installed.

I cannot investigate detailed reason. However I found a workaround. If I add `gem "sass-rails", "~> 5.0"` and `gem "refile", "~> 0.5"` to Gemfile, bundler installs the correct version of sass-rails.

Then all tests have been passed now.

**Comment by mshibuya** [comment]
Merged in, thanks :ok_woman:

**Pull request merged**